src/Core/Layer/Collector/ComposerFilesParser.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\Core\Layer\Collector;
+
+use JsonException;
+use RuntimeException;
+
+class ComposerFilesParser
+{
+    /**
+     * @var array{
+     *     packages?: array<string, array{
+     *          name: string,
+     *          autoload?: array{'psr-0'?: array<string, string>, 'psr-4'?: array<string, string>},
+     *          autoload-dev?: array{'psr-0'?: array<string, string>, 'psr-4'?: array<string, string>},
+     *     }>,
+     *     packages-dev?: array<string, array{
+     *          name: string,
+     *          autoload?: array{'psr-0'?: array<string, string>, 'psr-4'?: array<string, string>},
+     *          autoload-dev?: array{'psr-0'?: array<string, string>, 'psr-4'?: array<string, string>},
+     *     }>,
+     * }
+     */
+    private array $lockFile;
+
+    /**
+     * @var array<string, array{
+     *     autoload?: array{'psr-0'?: array<string, string>, 'psr-4'?: array<string, string>},
+     *     autoload-dev?: array{'psr-0'?: array<string, string>, 'psr-4'?: array<string, string>},
+     * }>
+     */
+    private array $lockedPackages;
+
+    /**
+     * @throws RuntimeException
+     */
+    public function __construct(string $lockFile)
+    {
+        $contents = file_get_contents($lockFile);
+        if (false === $contents) {
+            throw new RuntimeException('Could not load composer.lock file');
+        }
+        try {
+            /**
+             * @var array{
+             *     packages?: array<string, array{
+             *          name: string,
+             *          autoload?: array{'psr-0'?: array<string, string>, 'psr-4'?: array<string, string>},
+             *          autoload-dev?: array{'psr-0'?: array<string, string>, 'psr-4'?: array<string, string>},
+             *     }>,
+             *     packages-dev?: array<string, array{
+             *          name: string,
+             *          autoload?: array{'psr-0'?: array<string, string>, 'psr-4'?: array<string, string>},
+             *          autoload-dev?: array{'psr-0'?: array<string, string>, 'psr-4'?: array<string, string>},
+             *     }>,
+             * } $jsonDecode
+             */
+            $jsonDecode = json_decode($contents, true, 512, JSON_THROW_ON_ERROR);
```

Comment:
Probably not a problem in real use, but this constructor is quite heavy. It might make sense to lazily load and decode the JSON instead.

Reply:
Well the only place I instantiate this class I immediately use it (on the `ComposerCollector`), so it would make no difference.